cs-couchbase/src/main/java/io/cloudslang/content/couchbase/entities/inputs/BucketInputs.java

```diff
@@ -9,30 +9,195 @@
+    }
+
+    public String getReplicaIndex() {
+        return replicaIndex;
+    }
+
+    public String getSaslPassword() {
+        return saslPassword;
+    }
+
+    public Integer getProxyPort() {
+        return proxyPort;
+    }
+
+    public Integer getRamQuotaMB() {
+        return ramQuotaMB;
+    }
+
+    public Integer getReplicaNumber() {
+        return replicaNumber;
+    }
+
+    public Integer getThreadsNumber() {
+        return threadsNumber;
+    }
+
     public static class Builder {
+        private String authType;
         private String bucketName;
+        private String bucketType;
+        private String conflictResolutionType;
+        private String evictionPolicy;
+        private String flushEnabled;
+        private String parallelDBAndViewCompaction;
+        private String replicaIndex;
+        private String saslPassword;
+
+        private Integer proxyPort;
+        private Integer ramQuotaMB;
+        private Integer replicaNumber;
+        private Integer threadsNumber;
 
         public BucketInputs build() {
             return new BucketInputs(this);
         }
 
-        public BucketInputs.Builder withBucketName(String inputValue) {
-            bucketName = inputValue;
+        public BucketInputs.Builder withAuthType(String input) {
+            authType = AuthType.getAuthType(input);
+            return this;
+        }
+
+        public BucketInputs.Builder withBucketName(String input) {
+            bucketName = input;
+            return this;
+        }
+
+        public BucketInputs.Builder withBucketType(String input) {
+            bucketType = BucketType.getBucketType(input);
+            return this;
+        }
+
+        public BucketInputs.Builder withConflictResolutionType(String input) {
+            conflictResolutionType = ConflictResolutionType.getConflictResolutionType(input);
+            return this;
+        }
+
+        public BucketInputs.Builder withEvictionPolicy(String input) {
+            evictionPolicy = EvictionPolicy.getEvictionPolicy(input);
+            return this;
+        }
+
+        public BucketInputs.Builder withFlushEnabled(String input) {
+            flushEnabled = getEnabledString(input, false);
+            return this;
+        }
+
+        public BucketInputs.Builder withParallelDBAndViewCompaction(String input) {
+            parallelDBAndViewCompaction = valueOf(getEnforcedBooleanCondition(input, Boolean.FALSE));
+            return this;
+        }
+
+        public BucketInputs.Builder withReplicaIndex(String input) {
+            replicaIndex = getEnabledString(input, true);
+            return this;
+        }
+
+        public BucketInputs.Builder withProxyPort(String input) {
+            proxyPort = getValidPort(input);
+            return this;
+        }
+
+        public BucketInputs.Builder withRamQuotaMB(String input) {
+            ramQuotaMB = getValidIntValue(input, MINIMUM_RAM_QUOTA_AMOUNT, null, MINIMUM_RAM_QUOTA_AMOUNT);
```

Comment:
I personally don't like the validations to be done inside the Builder, but outside ( before you construct the builder)
But you can leave it be because either way a large part of the content has the same pattern :)